platform/default/online_file_source.cpp

```diff
@@ -176,6 +167,66 @@ class OnlineFileSource::Impl {
         }
     }
 
+    struct PendingRequests {
+        // initialize two pending queues: a high priority queue for "regular" requests and a low priority queue for offline download requests
+        // the high priority requests are in prioritizedPendingRequests*[0]
+        // the low priority requests are in prioritizedPendingRequests*[1]
+        // context: https://github.com/mapbox/mapbox-gl-native/issues/12655
+        PendingRequests() : prioritizedPendingRequestsList(2), prioritizedPendingRequestsMap(2) {}
+
+        std::vector<std::list<OnlineFileRequest*>> prioritizedPendingRequestsList;
```

Comment:
Using `std::array<std::list<OnlineFileRequest*>, 2>` conveys the intention better: a `std::vector` is resizable and uses dynamically allocated memory, while a `std::array` is fixed size and is stored inline.